engine/script/src/script_json.cpp

```diff
@@ -45,18 +45,35 @@ namespace dmScript
      * @namespace json
      */
 
-    int JsonToLua(lua_State* L, const char* json, size_t json_len)
+    static int JsonToLuaInternal(lua_State* L, const char* json, size_t json_len, int protected_mode)
     {
         int top = lua_gettop(L);
-        int ret = lua_cjson_decode(L, json, json_len);
+        char buffer[256] = {0};
+        int ret = lua_cjson_decode(L, json, json_len, protected_mode, buffer, sizeof(buffer));
         if (ret != 1)
         {
             lua_pop(L, lua_gettop(L) - top);
         }
-        assert(top + 1 == lua_gettop(L));
+
+        if (protected_mode)
+        {
+            assert(top + 1 == lua_gettop(L));
+        }
+        else if(!protected_mode && (ret == 0)) // let's restore the stack if we couldn't create
+        {
+            int num_to_pop = lua_gettop(L) - top;
+            lua_pop(L, num_to_pop);
+
+            dmLogError("%s", buffer);
+        }
         return ret;
     }
 
+    int JsonToLua(lua_State* L, const char* json, size_t json_len)
+    {
+        return JsonToLuaInternal(L, json, json_len, 0);
+    }
```

Comment:
The public dmSdk function is still using Lua in an unprotected manner, but we have now also disabled the luaL_errors()